src/components/shared/Button.tsx

```diff
@@ -1,78 +1,79 @@
 import {
   ActivityIndicator,
   ImageSourcePropType,
+  StyleProp,
   TouchableOpacity,
   ViewStyle,
 } from 'react-native';
+import React, { ReactText } from 'react';
 
-import React from 'react';
 import styled from 'styled-components/native';
 
 interface StyledElement {
   white?: boolean;
   disabled?: boolean;
 }
 
-interface ButtonContainer {
-  width: number;
-  height: number;
+interface ButtonProps {
+  width?: ReactText;
+  height?: ReactText;
 }
 
 interface Props {
   testID: string;
   isWhite: boolean;
   isLoading: boolean;
   isDisabled: boolean;
-  containerStyle?: ViewStyle;
+  containerStyle?: StyleProp<ViewStyle>;
   style?: ViewStyle;
-  onPress: () => void;
+  onPress: (params?: any) => void | Promise<void>;
   imgLeftSrc?: ImageSourcePropType;
   indicatorColor: string;
   activeOpacity: number;
   children: string;
-  width: number;
-  height: number;
+  width?: ReactText;
+  height?: ReactText;
 }
 
-const StyledButtonContainer = styled.View<ButtonContainer>`
-  width: ${({ width }): string => `${width}px`};
-  height: ${({ height }): string => `${height}px`};
+const ButtonContainer = styled.View<StyledElement>`
+  flex: 1;
```

Comment:
@marsinearth, it isn't good idea to use `flex: 1` here. please check #31

Reply:
Could you remove `flex: 1` in the future PR please?

Reply:
this resolves in #52